### PR TITLE
[IDP-948] Added reusable workflow for Semgrep validation

### PR DESCRIPTION
**Pull request by heqianwang**
Added reusable workflow for Semgrep validation

### CODE REVIEW

File: .github/workflows/reusable-semgrep-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Semgrep
+
+on:
+  workflow_call:
+    inputs:
+      github_event_name:
+        required: true
+        type: string
+      github_pr_sha:
```
Comment (meziantou) [inline .github/workflows/reusable-semgrep-workflow.yml:9]: The name is not clear. Which sha is it? head sha, base sha, something else?

File: .github/workflows/reusable-semgrep-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: Semgrep
+
+on:
+  workflow_call:
+    inputs:
+      github_event_name:
```
Comment (meziantou) [inline .github/workflows/reusable-semgrep-workflow.yml:6]: Can the context be inherited?

Comment (heqianwang) [reply .github/workflows/reusable-semgrep-workflow.yml]: We can actually inherit the context here, removing them

File: .github/workflows/reusable-semgrep-workflow.yml (hunk 1 above)
Comment (heqianwang) [reply .github/workflows/reusable-semgrep-workflow.yml]: We can actually inherit the context here, removing them

**Review by Gcaya — APPROVED** [review]

**Pull request merged**